app/assets/stylesheets/component_guide/application.scss

```diff
@@ -1,6 +1,43 @@
 // This file contains the styles for the Component Guide.
 
-@import "govuk_publishing_components/all_components";
+// This is the file that the application needs to include in order to use
+// the components.
+
+// feature flag for accessible link styles
+$govuk-new-link-styles: true;
+
+@import "govuk_publishing_components/govuk_frontend_support";
+@import "govuk_publishing_components/component_support";
+
+// Include all govuk frontend components
+@import "govuk/components/all";
+
+// Include stylesheets used in static
```

Comment:
Why do we need stylesheets that are used in `static` specifically? I think it would be good to have the answer in the commit message for this file, to explain "required stylesheets".

Reply:
Oh, is it because the component guide (unlike pages served from the application) isn't built using static, and therefore the styles from static aren't available? That makes sense, thanks. Maybe worth expanding the comment to explain that?

Reply:
Yes, although it is possible to make the styles from static available and not import them into application.css, some components do not render correctly using this approach.

This appears to be because we always expect stylesheets to be loaded in a certain order, those included in static, followed by those in the rendering app. This can cause rendering issues in apps that do not use this approach, an example of this is shown in the visual regression test for the `textarea` component in this PR.

The component-guide currently loads the stylesheets in the order below:

```SCSS
...
- character-count
- ...
- textarea
```

Applies the CSS below:

`.gem-c-textarea .govuk-textarea {margin-bottom: 0;}`

The rendering apps, load them in the order below:

```SCSS
- textarea
- ...
- character-count
```

Applies the CSS below:
`.govuk-character-count .govuk-textarea {margin-bottom: 5px;}`

Importing the `character-count` stylesheet after the `textarea` stylesheet ensures any overrides work as expected.

Overall there is more work to do in this area, I have created a separate card to fully implement the Per Page Asset Loading feature in the component guide, as part of this we will need to identify the correct order for stylesheets to be loaded - https://trello.com/c/Hhs2H5S7/2056-fully-implement-individual-loading-of-assets-in-the-component-guide.

I've updated the comment in the `application.css` to provide further information on why this approach is used.

Reply:
Good stuff. Alarming that our CSS is apparently so brittle! We should definitely look into fixing this at some point.